server/controllers/medical/patients/visits.js

```diff
@@ -27,44 +29,91 @@ exports.detail = detail;
 exports.listByPatient = listByPatient;
 exports.admission = admission;
 exports.discharge = discharge;
+exports.transfer = transfer;
+exports.patientAdmissionStatus = patientAdmissionStatus;
 
 const COLUMNS = `
-  BUID(uuid) AS uuid, BUID(patient_uuid) as patient_uuid, start_date, start_notes,
-  end_date, end_notes, user_id, user.username, start_diagnosis_id, end_diagnosis_id,
-  ISNULL(end_date) AS is_open, icd10.label as start_diagnosis_label, icd10.code as start_diagnosis_code,
-  hospitalized
+  BUID(patient_visit.uuid) AS uuid, BUID(patient_visit.patient_uuid) as patient_uuid,
+  patient_visit.start_date, patient_visit.start_notes, patient_visit.end_date, patient_visit.end_notes,
+  patient_visit.user_id, user.username, patient_visit.start_diagnosis_id, patient_visit.end_diagnosis_id,
+  DATEDIFF(IFNULL(patient_visit.end_date, CURRENT_DATE()), patient_visit.start_date) AS duration,
+  ISNULL(patient_visit.end_date) AS is_open, icd10.label as start_diagnosis_label, icd10.code as start_diagnosis_code,
+  patient_visit.hospitalized,
+  z.bed_label, z.room_label, z.ward_name,
+  patient.display_name, patient.hospital_no, em.text AS reference,
+  s.name AS service_name,
+  dt.label AS discharge_label,
+  inside_health_zone, is_new_case, is_refered, is_pregnant, last_service_id, discharge_type_id
+`;
+
+const TABLES = `
+  patient_visit
```

Comment:
What is the goal of this SQL query?  I can't seem to figure out what it does.

Let's talk this through tomorrow.